src/Mod/TechDraw/Gui/ViewProviderDrawingView.cpp

```diff
@@ -477,3 +479,55 @@ TechDraw::DrawView* ViewProviderDrawingView::getViewObject() const
 {
     return dynamic_cast<TechDraw::DrawView*>(pcObject);
 }
+
+
+//! it can happen that child graphic items can lose their parent item if the
+//! the parent is deleted, then undo is invoked.  The linkages on the App side are
+//! handled by the undo mechanism, but the QGraphicsScene parentage is not reset.
+void ViewProviderDrawingView::fixSceneDependencies()
+{
+    Base::Console().Message("VPDV::fixSceneDependencies()\n");
+    auto page = getViewProviderPage();
+    if (!page) {
+        return;
+    }
+
+    auto scene = page->getQGSPage();
+    auto ourQView = getQView();
+
+    // this is the logic for items other than Dimensions and Balloons
+    auto children = getViewObject()->getUniqueChildren();

```

Comment:
While `auto` is easier got use when writing code, it makes it more difficult to read code where type is not obvious (including review as there's little to no help with determining the type).

https://freecad.github.io/DevelopersHandbook/bestpractices/c++practices.html#the-use-of-auto-is-ok-if-the-right-hand-side-makes-clear-which-type-it-is

Reply:
I'll try to remember this for next time.   The types are all obvious to me (due to excessive exposure).  :)